### ISSUE TITLE
Interactive Add does not work

### ISSUE THREAD

**Issue by vasylskorych** [gitextensions/gitextensions#11648]
### Environment

- Git Extensions 4.2.1.17611
- Build b0c0b284876b63c52c77195ada82a85bfadbb083
- Git 2.44.0.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.28
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

Interactive Add command in Commit window does not work. If i choose the command, a git window pops up for a moment and dissapears right away. The command 
`git add --patch "myfile"`
appears in the command log though. If I run the same command manually, everything works as expected.
Basically, it works sometimes, but most of the times, it doesn't. The only pattern I see, it starts working again after a new update arrives, but it doesn't last for long.

### Steps to reproduce

Open Commit window -> right click on any file -> Interactive Add -> nothing happens.

**Comment by vasylskorych** [comment]
I played around a little more. I'm using Git extensions from Visual Studio via the extension plugin (v3.4.1.21). I thought this was just adding a few buttons to the Visual Studio UI and running the same executables in the background. But there seems to be a dependency on how I run GE: opening the window from Visual Studio shows the error above, but running it manually causes the Interactive Add feature to work correctly.

**Comment by pmiossec** [comment]
It will require a little more investigation on your part to really understand the issue (using task manager or process explorer):
* Are exactly the same exe launched?
* if you launch GE with the exact same parameters than Visual Studio from command line, do you reproduce? Maybe a problem of working directory as the path passed is relative...

Side question`/work around: Is there a reason why you don't do the operation on chunks directly from GE UI? (Because I found that much more convenient and never used this feature. Select lines and stash with `S`, unstash with `U` and  reset with `R`)

**Comment by vasylskorych** [comment]
I can confirm that the same exe is running. Running from VS only adds a current file as the parameter:
![image](https://github.com/gitextensions/gitextensions/assets/4201436/2461bd9a-dbce-48fa-ba7f-918ac49b0179)
The first is from VS, the second is a manual run.

Running the first command from powershell shows the same behavior as the manual run - everything works as expected. So, probably, some wizzardy is happening on the VS side.

Thanks for the advice! I somehow was'n aware of this feature. Now I have two ways to workaround 😄

**Comment by pmiossec** [comment]
> Running the first command from powershell shows the same behavior as the manual run - everything works as expected. So, probably, some wizzardy is happening on the VS side.

For the 2 processes, you should have a look if the "Working directory" (could also been named "Current directory") is different. And when trying to reproduce with powershell launch from this directory found...

**Comment by vasylskorych** [comment]
Yes, here is exactly what I did using Process Explorer and powershell: 
Current directory for a manual run is obviously "C:\Program Files (x86)\GitExtensions\", and for the VS run it is a directory inside my repo, say "D:\Codes\Dyssol\Utilities". So, in powershell I did:
```
cd D:\Codes\Dyssol\Utilities
& "C:\Program Files (x86)\GitExtensions\GitExtensions.exe" browse "D:\Codes\Dyssol\Utilities\ContainerFunctions.h"
```
And it works.

Maybe it could be some interplay with environment variables set by VS? I can make a screenshot of them for both cases if it helps.

**Comment by vbjay** [comment]
My hunch is vs is doing stuff to get console output and messing with the console output on ge because of it.

**Comment by vasylskorych** [comment]
But what makes this command so special? It seems, "Reset chunk of file" shows the same behavior, but the rest (stage, reset, skip worktree) work just fine.

**Comment by vbjay** [comment]
Prob interactive vs non interactive commands.
